pymc3/distributions/bound.py

```diff
@@ -263,20 +81,39 @@ class Bound:
                     'par2', mu=0.0, sigma=1.0, initval=1.0)
     """
 
-    def __init__(self, distribution, lower=None, upper=None):
+    rv_op = boundrv
+
+    def __new__(cls, *args, **kwargs):
+        transform = kwargs.get("transform", UNSET)
+        base_rv_type = args[1].owner.op.dtype
+        if transform is UNSET and base_rv_type not in ["intX", "int32", "int64"]:
```

Comment:
Actually it seems that through the Distribution Metaclass magic we can check if a random Op is of type `Continuous`:

```python
isinstance(at.random.normal().owner.op, pm.Continuous)
```

Reply:
You should probably keep the old Exception if a transform is provided for a Continuous distribution

```python
 raise ValueError("Cannot transform a discrete variable")
```